cmd/access/main.go

```diff
@@ -103,6 +108,35 @@ func main() {
 			flags.StringVarP(&nodeInfoFile, "node-info-file", "", "", "full path to a json file which provides more details about nodes when reporting its reachability metrics")
 			flags.StringToIntVar(&apiRatelimits, "api-rate-limits", nil, "per second rate limits for Access API methods e.g. Ping=300,GetTransaction=500 etc.")
 			flags.StringToIntVar(&apiBurstlimits, "api-burst-limits", nil, "burst limits for Access API methods e.g. Ping=100,GetTransaction=100 etc.")
+			flags.BoolVar(&staked, "staked", false, "whether this node is a staked access node or not")
+			flags.StringVar(&stakedAccessNodeAddress, "staked-access-node-addr", "", "the address of the upstream staked access node if this is an unstaked access node")
+		}).
+		PreInit(func(node *cmd.FlowNodeBuilder) {
+			if staked {
+				return
+			}
+			// for an unstaked node, set the identity here explicitly since it will not be found in the protocol state
+			self := &flow.Identity{
+				NodeID:        node.NodeID,
+				NetworkPubKey: node.NetworkKey.PublicKey(),
+				StakingPubKey: nil,             // no staking key needed for the unstaked node
+				Role:          flow.RoleAccess, // unstaked node can only run as an access node
+				Address:       node.BaseConfig.BindAddr,
+			}
+			node.Me, err = local.New(self, nil)
```

Comment:
setting `node.Me` here makes sure that the FlowNodeBuilder does not attemps to initialize the node id from the protocol state.

Reply:
That's actually a detail I'd love to read as a code comment.